controllers/localvolumeset/localvolumeset_controller.go

```diff
@@ -87,7 +88,13 @@ func (r *LocalVolumeSetReconciler) reconcile(ctx context.Context, request reconc
 	}
 
 	// store a one to many association from storageClass to LocalVolumeSet
-	r.LvSetMap.DeregisterStorageClassOwner(lvSet.Spec.StorageClassName, request.NamespacedName)
+	r.LvSetMap.RegisterStorageClassOwner(lvSet.Spec.StorageClassName, request.NamespacedName)
+
+	// handle the LocalVolumeSet finalizer
+	err = r.syncFinalizer(*lvSet)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to update localvolumeset finalizer: %w", err)
+	}
```

Comment:
iirc - storageclass can't be deleted via ownerreferences and hence must be deleted manually?

Reply:
Currently, it has to be deleted manually. We can have it deleted by finalizer in a following PR?